### PR TITLE
ACTIN-4003: Keep PD-L1 IHC results that are both recent but with different measure in report

### PR DESCRIPTION
**Pull request by ninajacobs**
Fixes problem in https://hartwigmedical.atlassian.net/browse/ACTIN-4003:

### CODE REVIEW

File: report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ class IhcTestInterpreter {
     private val interpretationBuilder = IhcTestInterpretationBuilder()
 
     fun interpret(ihcTests: List<IhcTest>): List<IhcTestInterpretation> {
-        val latestIhcTestsByItem = ihcTests.groupBy { it.item }.map { (_, tests) ->
+        val latestIhcTestsByItem = ihcTests.groupBy { Pair(it.item, it.measure) }.map { (_, tests) ->
```
Comment (cbruel) [inline report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt]: minor but update val name to `latestIhcTestsByItemAndMeasure`?

Comment (cbruel) [reply report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt]: good to be aware that measure is also often null, so if you have one test with measure = null and one with measure = CPS, on the same date, both will be kept, but I think that's desired behaviour

**Review by cbruel — APPROVED** [review]

File: report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ class IhcTestInterpreter {
...
             tests.maxByOrNull { it.measureDate ?: LocalDate.MIN } ?: tests.first()
         }
         latestIhcTestsByItem.forEach(::interpret)
```
Comment (cbruel) [inline report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt]: bit unrelated to this PR but is there are reason why for ihc results with a scoreText we don't show the measure on the report (line 35), only for results with a score (line 36) we show the measure? Didn't look into it in detail, but I believe you could now get something like:
PD-L1 negative
PD-L1 negative

(while the first is TPS and second is CPS on the same date)

File: report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt (hunk 1 above)
Comment (ninajacobs) [reply report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt]: thanks, adding a test for it!

File: report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt (hunk 2 above)
Comment (ninajacobs) [reply report/src/main/kotlin/com/hartwig/actin/report/interpretation/IhcTestInterpreter.kt]: good point, I agree. Making a PR

**Review by cbruel — APPROVED** [review]
Nice!
